via/requests_tools/error_handling.py

```diff
@@ -1,27 +1,71 @@
 """Helpers for capturing requests exceptions."""
 
 from functools import wraps
+from logging import getLogger
 
-from requests import RequestException, exceptions
+import h_pyramid_sentry
+from requests import exceptions
 
 from via.exceptions import BadURL, UnhandledException, UpstreamServiceError
 
-REQUESTS_BAD_URL = (
-    exceptions.MissingSchema,
-    exceptions.InvalidSchema,
-    exceptions.InvalidURL,
-    exceptions.URLRequired,
-)
-REQUESTS_UPSTREAM_SERVICE = (
-    exceptions.ConnectionError,
-    exceptions.Timeout,
-    exceptions.TooManyRedirects,
-    exceptions.SSLError,
-)
+ERROR_MAP = {
+    # The user gave us a goofy URL
+    exceptions.MissingSchema: BadURL,
+    exceptions.InvalidSchema: BadURL,
+    exceptions.InvalidURL: BadURL,
+    exceptions.URLRequired: BadURL,
+    # We got something unexpected from the upstream server
+    exceptions.ConnectionError: UpstreamServiceError,
+    exceptions.Timeout: UpstreamServiceError,
+    exceptions.TooManyRedirects: UpstreamServiceError,
+    exceptions.SSLError: UpstreamServiceError,
+    # Anything else...
+    exceptions.RequestException: UnhandledException,
+}
 
 
-def _get_message(err):
-    return err.args[0] if err.args else None
+LOG = getLogger(__name__)
+
+
+def _summarise_requests_exception(
+    exception: exceptions.RequestException, error_message
+):
+    """Create a summary string from details from a `RequestsException`."""
+    url = f"'{exception.request.url}'" if exception.request else "unknown URL"
+    status = (
+        "no response"
+        if exception.response is None
+        else f"{exception.response.status_code}: '{exception.response.reason}'"
+    )
+    return f"Requests exception - {type(exception)} Got {status} from {url} ({error_message})"
```

Comment:
This error message on the end is often going to be pretty much what we logged, but it means we are independent of the format of requests exceptions messages.